searvey/_ndbc_api.py

```diff
@@ -0,0 +1,155 @@
+from searvey.utils import get_region
+
+logger = logging.getLogger(__name__)
+
+# Create an instance of NdbcApi
+ndbc_api = NdbcApi()
+
+
+@functools.lru_cache
+def _get_ndbc_stations() -> gpd.GeoDataFrame:
+    """
+    Return NDBC station metadata.
+    :return: ``geopandas.GeoDataFrame`` with the station metadata
+    """
+    stations_df = ndbc_api.stations()
+    stations_df[["lat", "ns", "lon", "ew"]] = stations_df["Location Lat/Long"].str.extract(
+        r"(\d+\.\d+)([N|S]) (\d+\.\d+)([E|W])"
+    )
+    stations_df["lat"] = pd.to_numeric(stations_df["lat"])
+    stations_df["lon"] = pd.to_numeric(stations_df["lon"])
+    stations_df["lat"] = stations_df["lat"] * np.where(stations_df["ns"] == "S", -1, 1)
+    stations_df["lon"] = stations_df["lon"] * np.where(stations_df["ew"] == "W", -1, 1)
+    stations_df = stations_df.drop(columns=["Location Lat/Long"])
+
+    stations_df = gpd.GeoDataFrame(
+        data=stations_df,
+        geometry=gpd.points_from_xy(stations_df.lon, stations_df.lat, crs="EPSG:4326"),
+    )
+    return stations_df
+
+
+def get_ndbc_stations(
+    region: MultiPolygon | Polygon | None = None,
+    lon_min: float | None = None,
+    lon_max: float | None = None,
+    lat_min: float | None = None,
+    lat_max: float | None = None,
+) -> gpd.GeoDataFrame:
+    """
+    Return NDBC station metadata.
+    If `region` is defined then the stations that are outside of the region are
+    filtered out. If the coordinates of the Bounding Box are defined then
+    stations outside of the BBox are filtered out. If both ``region`` and the
+    Bounding Box are defined, then an exception is raised.
+    :param region: ``Polygon`` or ``MultiPolygon`` denoting region of interest
+    :param lon_min: The minimum Longitude of the Bounding Box.
+    :param lon_max: The maximum Longitude of the Bounding Box.
+    :param lat_min: The minimum Latitude of the Bounding Box.
+    :param lat_max: The maximum Latitude of the Bounding Box.
+    :return: ``geopandas.GeoDataFrame`` with the station metadata
+    """
+    region = get_region(
+        region=region,
+        lon_min=lon_min,
+        lon_max=lon_max,
+        lat_min=lat_min,
+        lat_max=lat_max,
+        symmetric=True,
+    )
+
+    ndbc_stations = _get_ndbc_stations()
+    if region:
+        ndbc_stations = ndbc_stations[ndbc_stations.within(region)]
+    return ndbc_stations
+
+
+def _fetch_ndbc(
+    station_id: str,
+    mode: str,
+    start_time: pd.Timestamp,
+    end_time: pd.Timestamp,
+    columns: list[str] | None = None,
+) -> pd.DataFrame:
+    """Retrieve the TimeSeries of a single NDBC station."""
+    try:
+        df = ndbc_api.get_data(
+            station_id=station_id,
+            mode=mode,
+            start_time=start_time,
+            end_time=end_time,
+            as_df=True,
+            cols=columns,
+        )
+        if df.empty:
+            logger.warnings(f"No data avaliable for station {station_id}")
```

Comment:
I get precisely an error here: 
```
2024-07-03 18:09:49,110; ERROR   ; ThreadPoolExecutor-29_2; searvey._ndbc_api           105; Error fetching data for station 42059: 'Logger' object has no attribute 'warnings'```

`logger.warnings` does not exist

Reply:
Yea that error is logged if there isn't data available for that data mode in that specific station?

Reply:
Hi Abdu, no it's not what I meant. 
> logger.warnings does not exist

The `.warnings()` does not exist. `warning()` does ([ref](https://docs.python.org/3/library/logging.html#logging.Logger.warning)). There is the warnings [lib](https://docs.python.org/3/library/warnings.html#module-warnings) which is something else. 

Hence why 
```
'Logger' object has no attribute 'warnings'
```